File: Bakery.c
// Copyright (c) 2022 Nina Mathew All rights reserved
//
// Created by: Nina Mathew
// Created on: Mar 1, 2023
// This program the cost of croissants

#include <stdio.h>

int main()
{
    // This function calculates cost of croissants
    const float CROISSANTS = 2.55;
    const float HST = 0.13;

    float croissantNumberFloat = 0;
    int croissantNumber;
    float sub_total1;
    float sub_total2;
    float total1;
    float total2;

    // input
    printf("How many croissants would you like?: ");
    scanf("%f", &croissantNumberFloat);

    // process
    if (croissantNumberFloat == 0)
    {
        printf("Invalid input, please try again.\n");
    }
    if (croissantNumberFloat >= 6)
    {
        // output
        total1 = croissantNumberFloat * CROISSANTS;
        printf("The cost for the croissants is $ %.2f.\n", total1);
    }
    if (croissantNumberFloat < 6 && croissantNumberFloat > 0)
    {
        sub_total2 = croissantNumberFloat * CROISSANTS;
        total2 = sub_total2 + (sub_total2 * HST);
        printf("The cost for the croissants is $ %.2f.\n", total2);
    }
    printf("\nDone.\n");
    return 0;
}

File: bakery.py
#!/usr/bin/env python3
""""
Created by: Nina Mathew
Created on: Feb 28 2023
This is the "Calculates cost of a croissants" module
"""


import constants


def main():
    """The main() function calculates the cost of croissants, returns None."""
    # input
    croissant_number_str = input("Enter the number of croissants: ")
    # process
    try:
        croissant_number = int(croissant_number_str)
        if (
            croissant_number >= 6
        ):  # if the number of croissants is greater than or equal to 6
            total1 = constants.CROISSANT * croissant_number
            print(f"\nThe cost for a {croissant_number} croissants is ${total1:.2f}.")
        else:
            sub_total = constants.CROISSANT * croissant_number
            total2 = sub_total + (sub_total * constants.HST)
            print(f"\nThe cost for a {croissant_number} croissant's is ${total2:.2f}.")
    except ValueError:
        print("\nInvalid input")
    finally:
        print("\nThanks for coming! 😊.")


if __name__ == "__main__":
    main()
    print("\nDone")
